lib/executor/shared_iterations_test.go

```diff
@@ -0,0 +1,38 @@
+package executor
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/loadimpact/k6/lib/types"
+	"github.com/loadimpact/k6/stats"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	null "gopkg.in/guregu/null.v3"
+)
+
+func getTestSharedIterationsConfig() SharedIterationsConfig {
+	return SharedIterationsConfig{
+		VUs:         null.IntFrom(10),
+		Iterations:  null.IntFrom(100),
+		MaxDuration: types.NullDurationFrom(5 * time.Second),
+	}
+}
+
+func TestSharedIterationsRun(t *testing.T) {
+	t.Parallel()
+	var doneIters uint64
+	var ctx, cancel, executor, _ = setupExecutor(
+		t, getTestSharedIterationsConfig(),
+		func(ctx context.Context, out chan<- stats.SampleContainer) error {
+			atomic.AddUint64(&doneIters, 1)
+			return nil
+		},
+	)
+	defer cancel()
+	err := executor.Run(ctx, nil)
+	require.NoError(t, err)
+	assert.Equal(t, uint64(100), doneIters)
+}
```

Comment:
I would really prefer if there either an additional test or this is extend to test that if one VU is slow others will take more iterations.
Also for the PerVUIteration test, that if one VU is slow other VUs won't take iterations :)

Reply:
Aha, sure, makes sense. I'd rather do that in another test and leave this one as baseline.

I intentionally didn't add more tests here or in the other PR, as we need a _lot_ more of these more complex scenarios, and I just wanted to get the ball rolling. :)

But sure, I'll add that one test in this PR.

Reply:
Should be resolved by d70b41ea and c42edaf3.

I'm worried these might be flaky, but will address if it becomes a problem.